src/libcore/macros.rs

```diff
@@ -108,6 +108,44 @@ macro_rules! assert_eq {
     });
 }
 
+/// Asserts that two expressions are not equal to each other.
+///
+/// On panic, this macro will print the values of the expressions with their
+/// debug representations.
+///
+/// # Examples
+///
+/// ```
+/// let a = 3;
+/// let b = 2;
+/// assert_ne!(a, b);
+/// ```
+#[macro_export]
+///#[stable(feature = "????", since = "????")]
```

Comment:
macros are a bit odd, since they become insta-stable. Right now, `since` should be `1.12.0`. Unsure about `feature`, since it won't have one. @alexcrichton ?